CONTRIBUTING.md

```diff
@@ -0,0 +1,126 @@
+# Overview
+
+This documents explains the processes and practices recommended for
+contributing enhancements to the Prometheus charm.
+
+- Generally, before developing enhancements to this charm, you should consider
+  [opening an issue ](https://github.com/canonical/prometheus-operator) explaining
+  your the use case.
+- If you would like to chat with us about your use-cases or proposed
+  implementation, you can reach us at Canonical
+  [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev).
+- It is strongly recommended that prior to engaging in any enhancements
+  to this charm you familiarise your self with the Juju
+- Familiarising yourself with the
+  [Juju Operator SDK](https://juju.is/docs/sdk) and the Python
+  [Operator Framework](https://ops.readthedocs.io/en/latest/)
+  library will help you a lot when working on PRs.
+- All enhancements require review before being merged. Besides the
+  code quality and test coverage, the review will also take into
+  account the resulting user experience for Juju administrators using
+  this charm. Please help us out in having easier reviews by rebasing
+  onto the main branch, avoid merge commits and enjoy a linear Git
+  history.
+
+## Developing
+
+Create and activate a virtualenv with the development requirements:
+
+```bash
+$ virtualenv -p python3 venv
+$ source venv/bin/activate
+$ pip install -r requirements-dev.txt
+```
+
+### Setup
+
+A typical setup using [snaps](https://snapcraft.io/), for deployments
+to a [microk8s](https://microk8s.io/) cluster can be done using the
+following commands
+
+```bash
+$ sudo snap install microk8s --classic
+$ microk8s.enable dns
+$ sudo snap install juju --classic
+$ juju bootstrap microk8s microk8s
+$ juju create-storage-pool operator-storage kubernetes storage-class=microk8s-hostpath
+```
+
+### Build
+
+Install the charmcraft tool
+
+```bash
+$ sudo snap install charmcraft
+```
+
+Build the charm in this git repository
+
+```bash
+$ charmcraft build
+```
+
+### Deploy
+
+```bash
+$ juju deploy ./prometheus-k8s.charm --resource prometheus-image=ubuntu/prometheus:latest
+```
+
+## Testing
+
+Unit tests are implemented using the Operator Framework test
+[harness](https://ops.readthedocs.io/en/latest/#module-ops.testing). These
+tests may executed by doing
+
+```bash
+$ ./run_tests
+```
+
+## Code Overview
+
+The core implementation of this charm is represented by the
+[`PrometheusCharm`](src/charm.py) class. `PrometheusCharm` responds to
+configuation changes, changes in relations with Alertmanager, Grafana
+and any scrape target. In response to any change in its configuration,
+relations with Alertmanager or scrape target, `PrometheusCharm`
```

Comment:
This section could probably be restructured into a table/list of bullets indicating stimulus/response and be a little easier to read, but this is style only so feel free to ignore :)

Reply:
Restructured "stimulus" as bullet points. Split "response" into separate paragraphs (with no text changes) since this may grow over time to be too big to summarise into a few bullet points.